src/main/java/lotto/domain/PurchasedLottos.java

```diff
@@ -0,0 +1,39 @@
+package lotto.domain;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public record PurchasedLottos(List<Lotto> values) {
+
+    public PurchasedLottos(Lotto... inputs) {
+        this(List.of(inputs));
+    }
+
+    public PurchasedLottos {
+        validateNotBlank(values);
+    }
+
+    private void validateNotBlank(List<Lotto> inputs) {
+        if (inputs == null || inputs.isEmpty()) {
+            throw new IllegalArgumentException("구매한 로또는 1개 이상이어야 합니다.");
+        }
+    }
+
+    public LottoResult result(WinningLotto winningLotto) {
+        List<LottoRank> ranks = new ArrayList<>();
+
+        for (Lotto lotto : values) {
+            ranks.add(winningLotto.match(lotto));
+        }
+
+        return new LottoResult(ranks);
```

Comment:
```suggestion
        LottoResult result = LottoResult();

        for (Lotto lotto : values) {
            result.updateRank(winningLotto.match(lotto));
        }

        return result;
```
위와 같이 구현할 수도 있는데 어떤 방식으로 구현하는 것이 나을까?
각각의 장단점은?

Reply:
두 방식의 장단점을 비교해봤습니다.

기존 (record + List 전달)
- 장점: 불변 객체
- 단점: 집계 책임이 호출부에 분산

변경 (class + updateRank)
- 장점: 집계 책임이 내부에 응집, Tell Don't Ask 준수
- 단점: 가변 객체

불변성보다 책임 응집을 우선했습니다.